MIGRATING_to_v3.md

```diff
@@ -0,0 +1,70 @@
+# Migrate to Design System version 3
+
+## Benefits
+
+- Tree-shaking for the `core` package (`pcln-design-system`)
+- Tree-shaking for the `icons` package (`pcln-icons`)
+  - Can now use `import { TrendingUp, Timer } from 'pcln-icons'` import syntax
+- Tree-shaking for the `autocomplete` package (`pcln-autocomplete`)
+- Tree-shaking for the `modal` package (`pcln-modal`)
+- Tree-shaking for the `popover` package (`pcln-popover`)
+- Tree-shaking for the `slider` package (`pcln-slider`)
+
+## Breaking Changes
+
+### Core (`pcln-design-system`)
+
+- peerDependency increase: `pcln-icons >=3`
+- GreenButton and RedButton have been removed
+  - Use `<Button color='secondary'>...</Button` or `<Button color='error'>...</Button>` instead
+- OutlineButton `<OutlineButton />` has been removed
+  - Use `<Button variation='outline'>...</Button>` instead
+- `bg` prop has been marked as deprecated in favor of `color`
+- Icon `<Icon name="iconName" />` has been removed
+- IconButton
+  - `name`, `size` and `color` props are removed
+  - added `icon` prop of type node. You should pass the `size` and `color` props directly to the icon child:
+
+```
+import { IconButton } from 'pcln-design-system
+import { ThumbsUp } from 'pcln-icons
+
+<IconButton
+    icon={<ThumbsUp size={32} color='primary' />}
+/>
+```
+
+- Components that now take an `icon` node instead of an `iconName` string
+
+  - Icon
+  - Hug
+  - Banner
+  - Step
+  - Select
+
+- Text
+  - Added `textShadowSize` and `enableTextShadow` props
+  - Changed default fontSize behavior of Heading to be more inline with Heading styles
+
+### Icons (`pcln-icons`)
+
+- `main` and `module` fields in `package.json` now point to CommonJS (CJS) and ES Modules (ESM) builds, respectively.
+- Usage: `import { TrendingUp, Timer } from 'pcln-icons'`
+  - Imports from `pcln-icons/lib/` are no longer available. This previously contained a CJS build that was incompatible with tree-shaking and dead code elimination in bundlers.
```

Comment:
Ah this answers my question from before.  So if a host app consumes two common components, one which uses DS v2, and one which uses DS v3, what is the proper way to use pcln-icons?

Reply:
Ideally, the common components shouldn't have the DS as a dependency but as a peerDependency which means the host app's version will be the only one included. However, this does mean that common components will have to update to use the new methods of v3 before the host app can update.

Reply:
If there is no longer a pcln-icons/lib, could pcln-icons v3 have a lib folder that is an actually just an alias that points back to the root?

Reply:
That definitely seems feasible but at what point do we stop doing that? With a v4 release do we remove lib and now people have to do update everything or do we force people to update now???

Reply:
Due to the breaking changes, I don't think there is a way to use both in one application.

Reply:
To @James300 's point, maybe it would be better not to create a `/lib` in DS itself, but instead use Webpack aliases or a similar mechanism to point to `/cjs` instead.